addons/base_import/models/base_import.py

```diff
@@ -951,15 +951,21 @@ def parse_preview(self, options, count=10):
                 advanced_mode = has_relational_header or has_relational_match
 
             # Take the first non null value for each column to display example to user
+            # initially the first non null value will be displayed to the user
+            # on the hover of value it will preview the 5 values.
             column_example = []
             for column_index, _unused in enumerate(preview[0]):
+                vals = []
                 for record in preview:
                     if record[column_index]:
-                        column_example.append("%s%s" % (record[column_index][:50], "..." if len(record[column_index]) > 50 else ""))
+                        vals.append("%s%s" % (record[column_index][:50], "..." if len(record[column_index]) > 50 else ""))
+                    if len(vals) == 5:
                         break
-                # add a blank value if no example have been found at all for the current column
                 else:
-                    column_example.append("")
+                    # add a blank value if no example have been found at all for the current column
+                    if not vals:
```

Comment:
Could be merged with else no ? Making an 'elif' and gaining an indentation level